File: algorithm/src/leetcode/TwoSum_1.java
package leetcode;

// #1 TwoSum 
import java.util.*;

public class TwoSum_1 {

	public static void main(String[] args) {
//		int[] nums = {2, 7, 11, 15};
//		int target = 9;
		
		int[] nums = {3, 2, 4};
		int target = 6;
		
		System.out.println(Arrays.toString(twoSum(nums, target)));
	}
	
	public static int[] twoSum(int[] nums, int target) {
		int n = nums.length;
		for(int i=0; i<n; i++) {
			for(int j=0; j<i; j++) {
				int sum = nums[i] + nums[j];
				if(sum == target) {
					return new int[] {j, i};
				}
			}
		}
		
		return new int[] {0,1};
    }
}

File: algorithm/src/leetcode/ruby/TwoSum_1.rb
# #75 ruby Sort Colors

# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  dict = {}
  nums.each_with_index do |num, i|
    if dict[target - num]
      return dict[target - num], i
    end
    dict[num] = i
  end
end

nums = [2,7,11,15]
target = 9
p two_sum(nums, target)